pkg/upgrade/upgrade.go

```diff
@@ -21,11 +23,28 @@ func ManagedInstances(ctx context.Context, c client.Client, reader client.Reader
 
 	list := &v1.JaegerList{}
 	identity := viper.GetString(v1.ConfigIdentity)
-	opts := client.MatchingLabels(map[string]string{
+	opts := []client.ListOption{}
+	opts = append(opts, client.MatchingLabels(map[string]string{
 		v1.LabelOperatedBy: identity,
-	})
-	if err := reader.List(ctx, list, opts); err != nil {
-		return tracing.HandleError(err, span)
+	}))
+	if err := reader.List(ctx, list, opts...); err != nil {
```

Comment:
You can have a conditional here like you have in the `jaeger_controller`, checking for cluster-permissions before failing due to the lack of permissions.

Reply:
Sure i can add that. The reason I did not add that before is that we have upgrade check before the autodect, that viper config was not set in the beginning. But no warm to add